### PR TITLE
CXX-555 add test coverage for collection::distinct

### PR DESCRIPTION
**Pull request by amidvidy**
I'm not using the mocking framework here because there isn't a clean way to interpose distinct without
using private APIs of the C driver.

### CODE REVIEW

File: src/mongocxx/test/collection.cpp [hunk 1]
```diff
@@ -388,4 +393,57 @@ TEST_CASE("CRUD functionality", "[driver::collection]") {
...
+        document doc1{};
+        doc1 << "foo" << "baz" << "garply" << 1;
+        document doc2{};
+        doc2 << "foo" << "bar" << "garply" << 2;
+        document doc3{};
+        doc3 << "foo" << "baz" << "garply" << 2;
+        document doc4{};
+        doc4 << "foo" << "quux" << "garply" << 9;
+
+        bulk_write bulk{false /* unordered */};
```
Comment (acmorrow) [inline src/mongocxx/test/collection.cpp]: Off topic: Do we really just use raw bool's for the ordered-ness of a bulk_write? That makes me sad. Does CRUD spec allow us leeway here?

**Comment by acmorrow** [comment]
So this test will require a live mongod somewhere to run, unlike all of our other tests?

**Comment by amidvidy** [comment]
we already have a number of tests that require a live mongod (i.e. all the other tests in that file)

**Comment by acmorrow** [comment]
OK then. LGTM

File: src/mongocxx/test/collection.cpp (hunk 1 above)
Comment (amidvidy) [inline src/mongocxx/test/collection.cpp]: yep. It's pretty gross. An enum class would definitely be better

Comment (acmorrow) [inline src/mongocxx/test/collection.cpp]: Worth fixing?

**Pull request merged**